src/oemof_tabular_plugins/datapackage/post_processing.py

```diff
@@ -0,0 +1,339 @@
+import pandas as pd
+from datapackage import Package
+import oemof.solph as solph
+
+
+RAW_OUTPUTS = ["investments"]
+PROCESSED_RAW_OUTPUTS = ["flow_min", "flow_max", "aggregated_flow"]
+RAW_INPUTS = [
+    "marginal_cost",
+    "carrier_cost",
+    "capacity_cost",
+    "storage_capacity_cost",
+    "capacity",
+    "expendable",
+    "storage_capacity",
+    "min_capacity",
+    "max_capacity",
+    "efficiency",
+    "renewable_factor",
+    "emission_factor",
+]
+
+
+def compute_renewable_share(results_df, argument_names, col_name):
+    _check_arguments(results_df, argument_names, col_name)
+
+
+def compute_cO2_emissions(results_df, argument_names, col_name):
+    _check_arguments(results_df, argument_names, col_name)
+
+
+def _check_arguments(series, index_names, col_name):
+    """Check that all required argument are present in the DataFrame slice"""
+    for arg in index_names:
+        if arg not in series.index:
+            raise AttributeError(
+                f"The column {arg} is not present within the results DataFrame and is required to compute '{col_name}', listed in the calculations to be executed"
+            )
+
+
+def compute_variable_costs(results_df, argument_names, col_name):
+    """TODO write a docstring here"""
+    _check_arguments(results_df, argument_names, col_name)
+    return results_df.aggregated_flow * (
+        results_df.marginal_cost.astype("float")
+        + results_df.carrier_cost.astype("float")
+    )
+
+
+CALCULATED_OUTPUTS = [
```

Comment:
@ciaradunks - This is where one defines the calculations, note that if you want to add yours, you also have to define the function which is used under "operation"